test/src/mapper.test.js

```diff
@@ -1337,4 +1341,142 @@ describe("mapper", () => {
+                },
+                author: {
+                  type: "string",
+                  mapItems: ["author"],
+                },
+              },
+            },
+          },
+        },
+      };
+
+      const input = {
+        configuration: { books: ["Book 1", "Book 2", "Book 3"] },
+      };
+
+      const mapper = new Mapper({ plugins });
+      const result = mapper.map(input, template);
+
+      it("SHOULD return correct resolved value", () => {
+        expect(result.configuration.Books).toStrictEqual([
+          {
+            author: "Leo",
+            name: "Book 1",
+          },
+          {
+            author: "Leo",
+            name: "Book 2",
+          },
+          {
+            author: "Leo",
+            name: "Book 3",
+          },
+        ]);
+      });
+    });
+
+    describe("WHEN use the plugin with afterMapping hook", () => {
+      const plugins = [
+        {
+          id: "book_plugin", // books dataset transforming plugin
+          beforeMapping: (mapItems) => { // zip the books array and prices array and make them a new object array
+            const zipped = mapItems.books.map(function (e, i) {
+              return [e, mapItems.prices[i]];
+            });
+            return zipped.map((x) => ({
+              name: x[0],
+              price: x[1],
+            }));
+          },
+          afterMapping: (resolvedValue) => { // sort books by their price and add ranking number for them
+            resolvedValue.sort((x, y) => y.price - x.price);
+            return resolvedValue.map((x, index) => {
+              return {
+                rank: index + 1,
+                book: x.name,
+                price: x.price,
+              };
+            });
+          },
+        },
+        {
+          id: "upperCase_plugin", // string format plugin, convert string to upper-cased
+          afterMapping: (resolvedValue) => resolvedValue.toUpperCase(),
+        },
+      ];
+
+      const template = {
+        configuration: {
+          type: "object",
+          properties: {
+            BookPriceRanking: {
+              type: "set",
+              mapItems: ["configuration"],
+              properties: {
+                name: {
+                  type: "string",
+                  mapItems: ["name"],
+                  plugins: "upperCase_plugin",
+                },
+                price: {
+                  type: "number",
+                  mapItems: ["price"],
+                },
+              },
+              plugins: "book_plugin",
+            },
+          },
+        },
+      };
+
+      const input = {
+        configuration: {
+          books: ["Book 1", "Book 2", "Book 3"],
+          prices: [11.5, 20.8, 7.99],
+        },
+      };
+
+      const mapper = new Mapper({ plugins });
+      const result = mapper.map(input, template);
+
+      it("SHOULD return correct resolved value", () => {
+        expect(result.configuration).toStrictEqual({
```

Comment:
❓ I can't remember if `toStrictEqual` actually fails if the order is different, does it?

Reply:
No. The property order is not crucial for this function.